### ISSUE TITLE
Fatal error. System.AccessViolationException: Attempted to read or write protected memory.

### ISSUE THREAD

**Issue by enxas** [godotengine/godot#77343]
### Godot version

v4.0.3.stable.mono

### System information

Windows 10 22H2

### Issue description

Godot editor crashes on launch after I added this code to the `SpellBook.cs` file:
`[Export]
public NodePath[] NodePaths { get; set; }`

Error that I get in the console:
```
Fatal error. System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
   at Godot.NativeInterop.NativeFuncs.godotsharp_internal_script_debugger_send_error(Godot.NativeInterop.godot_string ByRef, Godot.NativeInterop.godot_string ByRef, Int32, Godot.NativeInterop.godot_string ByRef, Godot.NativeInterop.godot_string ByRef, Godot.NativeInterop.godot_bool, godot_stack_info_vector ByRef)
   at Godot.NativeInterop.ExceptionUtils.SendToScriptDebugger(System.Exception)
   at Godot.NativeInterop.ExceptionUtils.LogException(System.Exception)
   at Godot.Bridge.ScriptManagerBridge.GetPropertyDefaultValuesForType(System.Type, IntPtr, Void (IntPtr, Void*, Int32))   at Godot.Bridge.ScriptManagerBridge.GetPropertyDefaultValues(IntPtr, Void (IntPtr, Void*, Int32))
   at Godot.NativeInterop.NativeFuncs.godotsharp_internal_reload_registered_script(IntPtr)
   at Godot.Bridge.ScriptManagerBridge.CreateScriptBridgeForType(System.Type, Godot.NativeInterop.godot_ref*)
   at Godot.Bridge.ScriptManagerBridge.GetOrCreateScriptBridgeForType(System.Type, Godot.NativeInterop.godot_ref*)
   at Godot.Bridge.ScriptManagerBridge.GetOrCreateScriptBridgeForPath(Godot.NativeInterop.godot_string*, Godot.NativeInterop.godot_ref*)
```

![image](https://github.com/godotengine/godot/assets/8630984/d125afa9-e394-476b-9b89-2e9087a648f8)

My dotnet info:

```
dotnet --info
.NET SDK (reflecting any global.json):
 Version:   6.0.408
 Commit:    0c3669d367

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19045
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\6.0.408\

Host:
  Version:      7.0.5
  Architecture: x64
  Commit:       8042d61b17

.NET SDKs installed:
  3.1.426 [C:\Program Files\dotnet\sdk]
  5.0.303 [C:\Program Files\dotnet\sdk]
  5.0.403 [C:\Program Files\dotnet\sdk]
  5.0.407 [C:\Program Files\dotnet\sdk]
  5.0.408 [C:\Program Files\dotnet\sdk]
  5.0.416 [C:\Program Files\dotnet\sdk]
  6.0.408 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.21 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.24 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.26 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.10 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.16 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.16 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.21 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.24 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.26 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.10 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.16 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.16 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.21 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.24 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.26 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.9 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.10 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.12 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.16 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.16 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  arm64 [C:\Program Files\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\arm64\InstallLocation]
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
```

### Steps to reproduce

Open MRP with Godot editor, build it then Run the project

### Minimal reproduction project

[MRP.zip](https://github.com/godotengine/godot/files/11530846/MRP.zip)

**Comment by enxas** [comment]
Workaround for my problem is instead of this:
```
[Export] 
public NodePath[] NodePaths { get; set; }
```
use this:
```
[Export]
public Godot.Collections.Array<NodePath> NodePaths { get; set; }
```

**Comment by RedworkDE** [comment]
> When I run MPR using Godot 4.0.3.mono from official Godot website it doesn't throw error, but when I run it with my built from source Godot 4.0.3 with fix applied, then it crashes. So the fix fixes one error and causes another.

This has nothing to do with the patch and everything with the fact that you built it yourself, the 4.0.3 commit will almost certainly also crash if you build it yourself.

Can you post the exact commands you have used to build godot & post the compiler version etc you used to build?

**Comment by enxas** [comment]
I pretty much followed Godot docs when compiling for dotnet:

`cd Desktop`
`git clone --depth 1 --branch 4.0.3-stable https://github.com/godotengine/godot.git`
`cd godot`
apply the fix from #75188
`scons p=windows target=editor module_mono_enabled=yes`
`bin/godot.windows.editor.x86_64.mono --generate-mono-glue modules/mono/glue`
`./modules/mono/build_scripts/build_assemblies.py --godot-output-dir=./bin --godot-platform=windows`
(I ran this command second time, because running it first time I encountered this problem #75152)
`bin/godot.windows.editor.x86_64.mono --generate-mono-glue modules/mono/glue`

For compiler I have installed Microsoft Visual Studio Community 2019 Version 16.11.26
MSVC v142 C++ compiler
